pkg/policy/policy.go

```diff
@@ -162,7 +193,7 @@ func (c *Client) NeedsUpdate(ctx context.Context, registryOpts types.RegistryOpt
 		return false, nil
 	}
 
-	c.populateOCIArtifact(ctx, registryOpts)
+	c.populateOCIArtifact(ctx, "", registryOpts)
```

Comment:
The artifact for the default repository will be initialized here, even if the user has overridden the repositories.